src/protocols/output_management.rs

```diff
@@ -0,0 +1,785 @@
+                        }
+                    } else {
+                        error!("EnableHead: Missing conf data");
+                    }
+                } else {
+                    error!("EnableHead: Missing client data");
+                }
+            }
+            zwlr_output_configuration_v1::Request::DisableHead { head } => {
+                if outdated {
+                    return;
+                }
+                if let Some(head_name) = head.data::<String>() {
+                    if let Some(output) = conf_data.new_out.get_mut(head_name) {
+                        output.current_mode = None;
+                        output.logical = None;
+                    }
+                } else {
+                    error!("DisableHead: missing attached output head name");
+                    return;
+                }
+            }
+            zwlr_output_configuration_v1::Request::Apply => {
+                if outdated {
+                    return;
+                }
+                if conf_data.serial != g_state.serial {
+                    return;
+                }
+                let mut new_conf = Vec::with_capacity(conf_data.new_out.len());
+                for (_, out) in conf_data.new_out.iter() {
+                    new_conf.push(match (out.current_mode, out.logical) {
+                        (Some(current_mode), Some(logical)) => niri_config::Output {
+                            off: !(out.current_mode.is_some() && out.logical.is_some()),
+                            name: out.name.clone(),
+                            scale: Some(logical.scale),
+                            transform: logical.transform,
+                            position: Some(niri_config::Position {
+                                x: logical.x,
+                                y: logical.y,
+                            }),
+                            mode: Some(niri_ipc::ConfiguredMode {
+                                width: logical.width as u16,
+                                height: logical.height as u16,
+                                refresh: Some(out.modes[current_mode].refresh_rate as f64 / 1000.),
+                            }),
+                            variable_refresh_rate: false,
+                        },
+                        _ => niri_config::Output {
+                            off: !(out.current_mode.is_some() && out.logical.is_some()),
+                            name: out.name.clone(),
+                            scale: None,
+                            transform: Transform::Normal,
+                            position: None,
+                            mode: None,
+                            variable_refresh_rate: false,
+                        },
+                    });
+                }
+                state.apply_new_conf(new_conf);
+                conf.succeeded();
+            }
+            zwlr_output_configuration_v1::Request::Test => conf.succeeded(),
```

Comment:
Did you look at cosmic-comp how hard it would be to do at least some of this?

Reply:
So from what i see:
- Verify the existence and validity of every head (mode...)
- They dont allow all the user to have no enabled outputs
- check if all the outputs have surfaces

Reply:
It check for at least one enabled output now